lib/puppet/parser/functions/type3x.rb

```diff
@@ -21,7 +21,7 @@ module Puppet::Parser::Functions
 
     klass = value.class
 
-    unless [TrueClass, FalseClass, Array, Bignum, Fixnum, Float, Hash, String].include?(klass) # rubocop:disable Lint/UnifiedInteger
+    unless %w[TrueClass FalseClass Array Bignum Fixnum Float Hash String].include?(klass.to_s)
```

Comment:
We should probably add Integer to this list since Bignum and Fixnum are indeed deprecated. When the deprecation warning is bypassed, we get the Puppet::ParseError instead because this function doesn't know what an Integer is.